pkg/dockerregistry/server/tagservice.go

```diff
@@ -150,39 +152,38 @@ func (t tagService) Lookup(ctx context.Context, desc distribution.Descriptor) ([
 }
 
 func (t tagService) Tag(ctx context.Context, tag string, dgst distribution.Descriptor) error {
-	imageStream, err := t.repo.imageStreamGetter.get()
-	if err != nil {
-		context.GetLogger(ctx).Errorf("error retrieving ImageStream %s/%s: %v", t.repo.namespace, t.repo.name, err)
-		return distribution.ErrRepositoryUnknown{Name: t.repo.Named().Name()}
-	}
+	dockerImageReference := fmt.Sprintf("%s/%s/%s@%s", t.repo.config.registryAddr, t.repo.namespace, t.repo.name, dgst.Digest.String())
 
-	image, err := t.repo.registryOSClient.Images().Get(dgst.Digest.String(), metav1.GetOptions{})
-	if err != nil {
-		context.GetLogger(ctx).Errorf("unable to get image: %s", dgst.Digest.String())
-		return err
-	}
-	image.SetResourceVersion("")
-
-	if !t.repo.config.pullthrough && !isImageManaged(image) {
-		return distribution.ErrRepositoryUnknown{Name: t.repo.Named().Name()}
-	}
-
-	ism := imageapiv1.ImageStreamMapping{
+	isi := imageapiv1.ImageStreamImport{
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace: imageStream.Namespace,
-			Name:      imageStream.Name,
+			Namespace: t.repo.namespace,
+			Name:      t.repo.name,
 		},
-		Tag:   tag,
-		Image: *image,
+		Spec: imageapiv1.ImageStreamImportSpec{
+			Import: true,
+			Images: []imageapiv1.ImageImportSpec{
+				{
+					From: kapi.ObjectReference{Kind: "DockerImage", Name: dockerImageReference},
+					To:   &kapi.LocalObjectReference{Name: tag},
+
+					ImportPolicy: imageapiv1.TagImportPolicy{
+						Insecure: true,
```

Comment:
?

Reply:
@smarterclayton At the moment the registry does not know the correct value of this flag.

Reply:
The registry knows if it's secure or not (for managed images). And source image streams have TagImportPolicy as well - it can be inherited for external images. It's not bulletproof though.

Reply:
How does the registry know if it's secure or not? I'm pretty sure it doesn't.

Reply:
> How does the registry know if it's secure or not? I'm pretty sure it doesn't.

You're right. I took into consideration just `passthrough` tls termination where the certificate paths are passed to registry via environment variables. For `edge` tls termination, the registry cannot be sure for sure.

Reply:
But that's just another reason for me for not replacing imagestreammapping with imagestreamimport.